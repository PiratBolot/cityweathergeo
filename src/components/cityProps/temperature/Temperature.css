.temperature_c {
    color: #435257;
    font-size: 40px;
    padding: 0 5px;
    line-height: 1;
}