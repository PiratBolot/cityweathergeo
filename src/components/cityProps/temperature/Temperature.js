import React from 'react';
import './Temperature.css';

class Temperature extends React.Component {
    render = () => (
        <div className="temperature_c">
            <span>{(this.props.temp - 273.14).toFixed(1) + "Â°C"}</span>
        </div>
    )
}

export default Temperature;