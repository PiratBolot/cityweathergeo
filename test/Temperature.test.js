import React from 'react';
import { shallow } from '../enzyme/enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Temperature from "../src/components/cityProps/temperature/Temperature";

describe('Temperature', () => {
    const temperature = 276.15;

    it('should render correctly', () => {
        const output = shallow(
            <Temperature temp={temperature} />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});