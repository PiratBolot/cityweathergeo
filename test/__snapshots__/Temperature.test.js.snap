// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature should render correctly 1`] = `
<div
  className="temperature_c"
>
  <span>
    3.0Â°C
  </span>
</div>
`;
